run-tests.pl

```diff
@@ -549,17 +552,29 @@ sub _run_test
 #
 # returns a pair [$res, $reason] where $res is one of:
 #  1 - success
+#  0 - skipped
 #  undef - failure
 #
 sub _run_test0
 {
    my ( $t, $test, $req_futures ) = @_;
+   my $skip_reason;
 
    my $success = eval {
       my @reqs;
       my $f_setup = Future->needs_all( @$req_futures )
          ->on_done( sub { @reqs = @_ } )
-         ->on_fail( sub { die "fixture failed - $_[0]\n" } );
+         ->on_fail( sub {
+            my ( $reason ) = @_;
+
+            # if any of the fixtures failed with a special 'SKIP' result, then skip
+            # the test rather than failing it.
+            if ( $reason =~ /^SKIP(: *(.*))?$/ ) {
+               my $r = ": $2" // '';
```

Comment:
The string `": $2"` is never going to be undefined here even if `$2` was. You probably want to use ternary instead

    my $r = defined $2 ? ": $2" : "";